profiling/src/profile/mod.rs

```diff
@@ -561,6 +662,50 @@ impl Profile {
             .get(&local_root_span_id)
             .map(i64::clone))
     }
+
+    fn upscale_values(&self, values: &[i64], labels: &[Label]) -> anyhow::Result<Vec<i64>> {
+        let mut new_values = values.to_vec();
+
+        if !self.upscaling_rules.is_empty() {
```

Comment:
nitpick (non-blocking): I personally prefer the use of guard clauses or some call it early-exit. Then the rest of the code would not be indented that far. It is just a personal preference, not hard feelings, feel free to ignore.

```suggestion
        if self.upscaling_rules.is_empty() {
            Ok(new_values)
        }
```

Reply:
For me it depends of the cases, in this case I would have to return `Ok(new_values)` at two callsites (where you spot it and at the end).
Which means, potential error (if there is a change and on a callsite you have `Ok(new_values)` and the other you have `Ok(values)` )